### DISCUSSION TITLE
Is there a way to create resource using modal?

### DISCUSSION THREAD

**Discussion by michaelgatuma** [filamentphp/filament#1315] [Help]
I want to use the page ButtonAction to open a  modal and create my resource there instead of opening an entire page just for that.

**Comment by danharrin** [comment]
Yeah this is possible, by adding an action to the List page of the resource.

https://filamentadmin.com/docs/2.x/admin/pages#modals

**Reply by michaelgatuma** [reply]
@danharrin thanks for your response.  I have been doing that. I guess then there is a bug or I am making some silly mistake. I am getting this error: 

> TypeError
> Filament\Pages\Actions\ButtonAction::action(): Argument #1 ($action) must be of type ?string, Closure given, called in C:\laragon\www\kris-v2\app\Filament\Resources\ProjectResource\Pages\ListProjects.php on line 31 (View: C:\laragon\www\kris-v2\resources\views\vendor\filament\components\page.blade.php) 

I am using v2, you can reproduce with this:

```
use App\Models\User;
use Filament\Pages\Actions\ButtonAction;
...
protected function getActions(): array
    {
        return [
            ButtonAction::make('delete')
                ->action(fn () => User::first()->delete())
                ->requiresConfirmation()
                ->modalHeading('Delete User')
                ->modalSubheading('Are you sure you\'d like to delete this user? This cannot be undone.')
                ->modalButton('Yes, delete User')
        ];
    }
...
```

**Reply by michaelgatuma** [reply]
I have upgraded my filament from 2.8.3 to 2.9.9, now I am getting a new error but atleast the page loads and the error only triggers when I click the ActionButton:
```
ErrorException
Undefined array key "mountedActionForm" 
```

**Reply by michaelgatuma** [reply]
> I have upgraded my filament from 2.8.3 to 2.9.9, now I am getting a new error but atleast the page loads and the error only triggers when I click the ActionButton:
> ```
> ErrorException
> Undefined array key "mountedActionForm" 
> ```

@danharrin

**Reply by danharrin** [reply]
Fixed in the next release.

**Comment by michaelgatuma** [comment]
Okay thank you very much
